azkaban-common/src/main/java/azkaban/project/JdbcProjectHandlerSet.java

```diff
@@ -333,4 +333,26 @@ public Integer handle(final ResultSet rs) throws SQLException {
       return rs.getInt(1);
     }
   }
+
+  public static class FlowFileResultHandler implements ResultSetHandler<List<byte[]>> {
+
+    public static String SELECT_FLOW_FILE =
+        "SELECT flow_file FROM project_flow_files WHERE "
+            + "project_id=? AND project_version=? AND flow_name=? AND flow_version=?";
+
+    @Override
+    public List<byte[]> handle(final ResultSet rs) throws SQLException {
+      if (!rs.next()) {
+        return Collections.emptyList();
+      }
+
+      final ArrayList<byte[]> data = new ArrayList<>();
```

Comment:
why not while(rs.next()) {
},
then we don't need to have the !rs.next() check in the first place.

Reply:
The difference between the two approach is that one will return Collections.emptyList and the other will return new ArrayList<>() if the query result is empty. And there is some performance advantage of returning emptyList:
https://stackoverflow.com/questions/5552258/collections-emptylist-vs-new-instance
Also, it is a commonly used approach in our current code, so I just followed the convention.